### DISCUSSION TITLE
How to create a DHT node suing version 1.1.1-1+b1

### DISCUSSION THREAD

**Discussion by leoncwu** [arvidn/libtorrent#7388] [Q&A]
I can only use libtorrent version 1.1.1-1+b1 in Debian 9 system. I am not able to build the latest version due to many reasons that I do not have control over. I hope create a DHT node so that I do not need to use a tracker. I did some investigation and put together the following solution, but it cannot pass the compilation.

`settings_pack settings;`
`settings.set_bool(settings_pack::enable_dht, true);`
`ession s(settings);`
`dht_settings dht;`
`dht.privacy_lookups = true;`
`dht.service_port = 6881;`
`s.set_dht_settings(dht);`
`s.start_dht();`
`int dht_port = s.listen_port();`
`std::cout << "DHT node listening on port " << dht_port << std::endl;`

The compiler complains about the deprecated function `s.start_dht()`. I am not able to find an alternative way to do this. Can anyone offer some help?

Thank you in advance for your help.

**Comment by arvidn** [comment]
create a `settings_pack` object ([docs](https://libtorrent.org/reference-Settings.html#settings_pack)) and set `enable_dht` to `true` ([docs](https://libtorrent.org/reference-Settings.html#enable_dht)). pass this settings pack into the `session` constructor.
